packages/renderer/src/lib/pod/PodsList.spec.ts

```diff
@@ -607,3 +607,25 @@ test('Expect to see empty page and no table when no container engine is running'
   const noContainerEngine = screen.getByText('No Container Engine');
   expect(noContainerEngine).toBeInTheDocument();
 });
+
+test('Expect environment column sorted by engineId', async () => {
+  getProvidersInfoMock.mockResolvedValue([provider]);
+
+  const podA = { ...pod1, Name: 'pod-aaa', engineId: 'engine-zzz', engineName: 'name-aaa' };
+  const podB = { ...pod2, Name: 'pod-bbb', engineId: 'engine-aaa', engineName: 'name-zzz' };
+
+  listPodsMock.mockResolvedValue([podA, podB]);
+  window.dispatchEvent(new CustomEvent('provider-lifecycle-change'));
+  window.dispatchEvent(new CustomEvent('extensions-already-started'));
+
+  await vi.waitUntil(() => get(providerInfos).length === 1 && get(podsInfos).length === 2, { timeout: 5000 });
```

Comment:
```suggestion
  await waitFor(() => {
    expect(get(providerInfos)).toHaveLength(1);
    expect(get(podsInfos)).toHaveLength(2);
  });
```

Might be nice to have waitFor everywhere, for consistency 👍